GameEngine/Services/AnimalActions/AnimalActions.cs

```diff
@@ -0,0 +1,325 @@
+﻿using GameEngine.Entities;
+using Repository;
+
+namespace GameEngine.Services
+{
+    /// <summary>
+    /// The class contains all common logic of the animals' behaviour.
+    /// </summary>
+    public class AnimalActions : IAnimalActions
+    {
+        private Random _random = new Random();
+
+        /// <summary>
+        /// Adds new animal to the list of the animals on the current board. 
+        /// </summary>
+        /// <param name="newAnimal">New animal.</param>
+        /// <param name="animals">Animals.</param>
+        /// <param name="board">Board.</param>
+        public void AddAnimal(IAnimal newAnimal, List<IAnimal> animals, Board board)
+        {
+            List<NewAnimalCoordinates> freeCells = CalculateFreeCellsToAddAnimal(animals, board);
+            if (board.GameBoard.Length > board.GameBoard.Length * ConstantsRepository.HalfOfBoard)
+            {
+                AddAnimalToList(newAnimal, freeCells, animals);
+            }
+            else
+            {
+                return;
+            }
+        }
+
+        /// <summary>
+        /// Adds animal to list with the new coordinates.
+        /// </summary>
+        /// <param name="newAnimal">New animal.</param>
+        /// <param name="newXCoordinate">New X coordinate.</param>
+        /// <param name="newYCoordinate">New Y coordinate.</param>
+        /// <param name="animals">Animals.</param>
+        private void AddAnimalToList(IAnimal newAnimal,List<NewAnimalCoordinates> freeCells, List<IAnimal> animals)
```

Comment:
Parameter comments have newXCoordinate and newYCoordinate but this method does not have these input parameters.